app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt

```diff
@@ -144,7 +145,37 @@ interface MetricsService {
     fun shouldTrack(event: Event): Boolean
 }
 
-class Metrics(private val services: List<MetricsService>, private val isTelemetryEnabled: () -> Boolean) {
+interface MetricController {
+    fun start()
+    fun stop()
+    fun track(event: Event)
+
+    companion object {
+        fun create(services: List<MetricsService>, isTelemetryEnabled: () -> Boolean): MetricController {
+            return if (BuildConfig.TELEMETRY) return DebugMetricController()
```

Comment:
Is this condition reversed? It seems like a variable named BuildConfig.TELEMETRY would be true if telemetry is switched on.

Reply:
good catch, reversed it to make it more readable but forgot to move the objects :man_facepalming: